sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -269,50 +278,57 @@ object FileFormatWriter extends Logging {
   }
 
   /**
+   * For every registered [[WriteJobStatsTracker]], call `processStats()` on it, passing it
+   * the corresponding [[WriteTaskStats]] from all executors.
+   */
+  private def processStats(
+      statsTrackers: Seq[WriteJobStatsTracker],
+      statsPerTask: Seq[Seq[WriteTaskStats]])
+    : Unit = {
+
+    val statsPerTracker = if (statsPerTask.nonEmpty) {
+      statsPerTask.transpose
+    } else {
+      statsTrackers.map(_ => Seq.empty)
+    }
+    assert(statsTrackers.length == statsPerTracker.length,
+      s"""Every WriteTask should have produced one `WriteTaskStats` object for every tracker.
+         |statsTrackers = ${statsTrackers}
+         |statsPerTracker = ${statsPerTracker}
+     """.stripMargin)
```

Comment:
In case of many states, this might result big error message. Just printing length is enough?